### PR TITLE
Remove unnecessary and incorrect pkgconfig changes

### PR DESCRIPTION
**Pull request by HiassofT**
Commit https://github.com/LibreELEC/LibreELEC.tv/commit/ebd0c203b88116c7a553278f41aa0ef3e76f6ba7 broke keyboard selection in kodi on RPi4 (and likely all other non-x11 builds) as it incorrectly moved the xkb files from /usr/share/X1/xkb1 to /usr/lib/X11/xkb and thus kodi couldn't find them anymore
```
warning <general>: CLibInputSettings: unable to open: /usr/share/X11/xkb/rules/base.xml
```

The fix for that particular issue has now been moved to #7092 so this PR only contains the other pkgconfig changes that aren't actually needed.

The xkeyboard-config patches that followed that commit and tried to fix X11 builds were only workarounds for the self-inflicted problem caused by changing datadir (from the default share, to lib) and didn't fully succeed as all gbm/libinput builds were still broken

The root cause was trying to fix something that wasn't broken - both the arch specific /usr/lib/pkgconfig and the arch independent /usr/share/pkconfig directories exist for a reason and eg xkeyboard-config  (which is an arch independent data package) did the right thing and installed the pkgconfig to /usr/share/pkgconfig - trying to change that to /usr/lib is just wrong and has caused a lot of hassles.

The same is true for the other packages which were patched in #6175 , respect their decision, systemd eg was very clear about their choice, and don't try to be clever - so revert all the additional patches and non-standard config options, that makes life a lot easier for us.

Runtime tested on RPi4, please build and runtime test on other platforms

### CODE REVIEW

**Comment by rellla** [comment]
Not 100% sure, but should the keyboard problem be solved with #7092 ? 

I've still no keyboard working in kodi and the following in the logs:
`Nov 13 19:07:43.477866 LibreELEC kodi.sh[1277]: xkbcommon: ERROR: failed to add default include path /usr/lib/X11/xkb`
There is no /usr/lib/X11 directory. `libxkbcommon.so` is located in /usr/lib.

**Comment by heitbaum** [comment]
Hi @HiassofT - I’ve gone though this - revalidated my notes, tested and happy for the reverts for `systemd`, `util-macros`, `xtrans`, `xorgproto` - as these are genuinely architecture independant. Can you drop the revert for `zlib`, 1cf38cc0a9839803c312f2ed86ea52fb48735245 - as it is necessary - the `zlib.pc` is not architecture Independant.

File: packages/compress/zlib/package.mk [hunk 1]
```diff
@@ -11,7 +11,3 @@ PKG_DEPENDS_HOST="cmake:host"
 PKG_DEPENDS_TARGET="toolchain"
 PKG_LONGDESC="A general purpose (ZIP) data compression library."
 PKG_TOOLCHAIN="cmake-make"
-
```
Comment (heitbaum) [inline packages/compress/zlib/package.mk:14]: Please drop changes to packages/compress/zlib/package.mk. These are needed.

**Review by heitbaum — CHANGES_REQUESTED** [review]
Please drop 1cf38cc0a9839803c312f2ed86ea52fb48735245

File: packages/compress/zlib/package.mk (hunk 1 above)
Comment (HiassofT) [reply packages/compress/zlib/package.mk]: done

**Review by heitbaum — APPROVED** [review]

**Pull request merged**